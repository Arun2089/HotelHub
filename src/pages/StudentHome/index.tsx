import { useState, useEffect, useContext } from "react";
import { useNavigate } from "react-router-dom";
import { Box, Typography, Paper, Skeleton } from "@mui/material";
import {
  GridColDef,
  GridColumnHeaderParams,
  GridAlignment,
} from "@mui/x-data-grid";
import NoticeList from "@pages/Admin/notice/NoticeList";
import ErrorBoundary from "@components/ErrorBoundry";
import ErrorComponent from "@components/ErrorComponent";
import TableComponent from "@components/Table";
import ChipComponent from "@components/Chip";

import { AuthContext } from "@src/context/AuthContext";

import { NoticeDataType, ApplicationsType } from "@ts/types";
import {
  fetchData,
  extractArrayFromApiData,
  dateFormat,
  catchErrorMessage,
} from "@utils/index";
import {
  NOTICES_URL,
  COMPLAINTS_URL,
  ROOM_INFO_URL,
  STUDENT_ROOM_DETAILS,
  ERROR,
  APPLICATIONS_URL,
} from "@constant/index";
import { ComplaintType } from "@ts/types";
import {
  useFetchApplicationListData,
  useFetchComplaintListData,
  useFetchNoticeData,
} from "@src/queryHooks/query";
import useAlert from "@src/hooks/useAlert";
import AlertComponent from "@src/components/Alert";

const StudentHome = () => {
  const [notices, setNotices] = useState<NoticeDataType[] | null>(null);

  const [complaintData, setComplaintData] = useState<ComplaintType[] | null>(
    null
  );
  const [applicationData, setApplicationData] = useState<
    ApplicationsType[] | null
  >(null);
  const [roomDetails, setRoomDetails] = useState<{
    wing: string;
    floor: string;
    room: string;
  } | null>(null);
  const [roomDetailsError, setRoomDetailsError] = useState<boolean>(false);
  const {
    user: { studentInfo },
  } = useContext(AuthContext);
  const { alert, handleAlert } = useAlert();
  const { isOpen, message, severity } = alert;

  const navigate = useNavigate();

  const {
    data: complaintApiData,
    isLoading: isLoadingComplaintData,
    isError: compaintDataError,
    error: complaintDataErrorMessage,
    refetch: complaintDataRefetch,
  } = useFetchComplaintListData(
    `${COMPLAINTS_URL}?populate=*&filters[student][id][$eq]=${studentInfo?.id}&sort=createdAt:desc`
  );

  const {
    data: applicationApiData,
    isLoading: isLoadingApplicationData,
    isError: applicationDataError,
    error: applicationDataErrorMessage,
    refetch: applicationDataRefetch,
  } = useFetchApplicationListData(
    `${APPLICATIONS_URL}?populate=*&filters[student][id][$eq]=${studentInfo?.id}&sort=createdAt:desc`
  );

  const {
    data: noticeData,
    isError: noticeDataError,
    error: noticeDataErrorMessage,
    refetch: noticeDataRefetch,
  } = useFetchNoticeData(NOTICES_URL);

  useEffect(() => {
    getRoomDetails();
  }, []);

  useEffect(() => {
    if (complaintApiData) {
      const data = extractArrayFromApiData<ComplaintType>(
        complaintApiData.data
      );
      setComplaintData(data);
    }
  }, [complaintApiData]);

  useEffect(() => {
    if (applicationApiData) {
      const data = extractArrayFromApiData<ApplicationsType>(
        applicationApiData.data
      );
      setApplicationData(data);
    }
  }, [applicationApiData]);

  useEffect(() => {
    if (noticeData) {
      const data = extractArrayFromApiData<NoticeDataType>(noticeData.data);
      setNotices(data);
    }
  }, [noticeData]);

  const errors = [
    { error: compaintDataError, errorMessage: complaintDataErrorMessage },
    { error: applicationDataError, errorMessage: applicationDataErrorMessage },
    { error: noticeDataError, errorMessage: noticeDataErrorMessage },
  ];

  useEffect(() => {
    errors.forEach(({ error, errorMessage }) => {
      if (error) {
        handleAlert(true, catchErrorMessage(errorMessage), ERROR);
      }
    });
  }, [compaintDataError, applicationDataError]);

  async function getRoomDetails(): Promise<void> {
    try {
      const response = await fetchData(ROOM_INFO_URL);
      setRoomDetails(response.data.attributes.data);
      setRoomDetailsError(false);
    } catch (error) {
      console.error(error);
      setRoomDetailsError(true);
    }
  }

  const commonOptions: {
    sortable: boolean;
    align: GridAlignment;
    headerAlign: GridAlignment;
    renderHeader: (params: GridColumnHeaderParams) => JSX.Element;
  } = {
    sortable: false,
    align: "center",
    headerAlign: "center",
    renderHeader({ colDef: { headerName } }: GridColumnHeaderParams) {
      return <Typography className="font-semibold">{headerName}</Typography>;
    },
  };

  const complaintTablecolumns: GridColDef[] = [
    {
      field: "id",
      headerName: "Id",
      minWidth: 100,
      headerClassName: "pl-8",
      cellClassName: "pl-8 hover:cursor-pointer",
      ...commonOptions,
    },
    {
      field: "date",
      headerName: "Date",
      minWidth: 150,
      headerClassName: "pl-8",
      cellClassName: "pl-8 hover:cursor-pointer",
      ...commonOptions,
    },
    {
      field: "type",
      headerName: "Type",
      minWidth: 170,
      flex: 1,
      cellClassName: "hover:cursor-pointer",
      ...commonOptions,
    },
    {
      field: "status",
      headerName: "Status",
      minWidth: 170,
      flex: 1,
      cellClassName: "hover:cursor-pointer",
      ...commonOptions,
      renderCell({ row }) {
        const { status } = row as ComplaintType;
        return (
          <ChipComponent
            className="w-20 capitalize"
            text={status}
            type={status}
          />
        );
      },
    },
  ];

  const applicationTableColumn: GridColDef[] = [
    {
      field: "id",
      headerName: "Id",
      minWidth: 100,
      headerClassName: "pl-8",
      cellClassName: "pl-8 hover:cursor-pointer",
      ...commonOptions,
    },
    {
      field: "createdAt",
      headerName: "Created at",
      minWidth: 150,
      headerClassName: "pl-8",
      cellClassName: "pl-8 hover:cursor-pointer",
      renderCell: ({ row }) => dateFormat(row.createdAt || ""),
      ...commonOptions,
    },
    {
      field: "subject",
      headerName: "Subject",
      minWidth: 170,
      flex: 1,
      cellClassName: "hover:cursor-pointer",
      ...commonOptions,
    },
    {
      field: "status",
      headerName: "Status",
      minWidth: 170,
      flex: 1,
      cellClassName: "hover:cursor-pointer",
      ...commonOptions,
      renderCell({ row }) {
        const { status } = row as ComplaintType;
        return (
          <ChipComponent
            className="w-20 capitalize"
            text={status}
            type={status}
          />
        );
      },
    },
  ];

  return (
    <Box className="p-4 w-full xl:p-8 flex flex-col min-h-20 flex-grow-0 md:flex-grow md:min-h-1/2 overflow-scroll">
      <Box className="flex-col gap-y-4 flex md:flex-row justify-around gap-x-6 md:gap-y-0 h-1/2 pb-4 md:pb-6">
        <ErrorBoundary
          error={roomDetailsError}
          ErrorComponent={
            <Paper className="dashboard-paper ">
              <ErrorComponent
                className="w-full h-full"
                onSubmit={getRoomDetails}
                message="Error in fetching data"
                heading="Room information"
                headingClassName="text-xl md:text-2xl font-medium mx-8 my-4 mb-4"
              />
            </Paper>
          }
        >
          <Paper className="w-full pb-8 md:w-[48%] lg:w-1/2 px-0 h-1/2 md:h-full flex-grow overflow-hidden rounded-xl">
            <Typography className="mx-8 my-4 mb-2.5 text-xl md:text-2xl font-medium ">
              Room Information
            </Typography>
            <Box className="flex justify-evenly px-4 gap-2 md:gap-4 h-32 md:h-40 mt-12">
              {roomDetails ? (
                <>
                  {STUDENT_ROOM_DETAILS.map(({ name, Icon }) => {
                    return (
                      <Box
                        key={name}
                        className="flex-1 h-full flex flex-col gap-3 justify-center items-center rounded-xl bg-primary-light bg-opacity-10"
                      >
                        <Typography className="text-3xl md:text:4xl lg:text-5xl font-semibold">
                          {roomDetails[name as keyof typeof roomDetails]}
                        </Typography>
                        <Box className="flex gap-2 items-center">
                          <Icon
                            color="primary"
                            className="text-base md:text-2xl"
                          />
                          <Typography className="text-sm md:text-lg">
                            {name}
                          </Typography>
                        </Box>
                      </Box>
                    );
                  })}
                </>
              ) : (
                <Skeleton variant="rectangular" width="100%" height="100%" />
              )}
            </Box>
          </Paper>
        </ErrorBoundary>

        <ErrorBoundary
          error={applicationDataError}
          ErrorComponent={
            <Paper className="dashboard-paper ">
              <ErrorComponent
                className="w-full h-full"
                onSubmit={applicationDataRefetch}
                message="Error in fetching data"
                heading="Today's Menu"
                headingClassName="text-xl md:text-2xl font-medium mx-8 my-4 mb-4"
              />
            </Paper>
          }
        >
          <Paper className="w-full pb-8 md:w-[48%] lg:w-1/2 px-0 h-1/2 md:h-full flex-grow overflow-hidden rounded-xl">
            <Typography className="mx-8 my-4 mb-2.5 text-xl md:text-2xl font-medium ">
              My Application
            </Typography>
            <TableComponent
              columns={applicationTableColumn}
              isLoading={isLoadingApplicationData}
              rows={applicationData || []}
              tableClassName="px-4 pb-4 h-full border-0"
              onRowClick={() => {
                navigate("applications");
              }}
            />
          </Paper>
        </ErrorBoundary>
      </Box>
      <Box className="flex-col gap-y-4 flex md:flex-row justify-around gap-x-6 md:gap-y-0 h-1/2 pb-3">
        <ErrorBoundary
          error={noticeDataError}
          ErrorComponent={
            <Paper className="dashboard-paper">
              <ErrorComponent
                className="w-full h-full"
                onSubmit={noticeDataRefetch}
                message="Error in fetching data"
                heading="Notices"
                headingClassName="text-xl md:text-2xl font-medium mx-8 my-4 mb-4"
              />
            </Paper>
          }
        >
          <NoticeList notices={notices} />
        </ErrorBoundary>

        <ErrorBoundary
          error={compaintDataError}
          ErrorComponent={
            <Paper className="dashboard-paper ">
              <ErrorComponent
                className="w-full h-full"
                onSubmit={complaintDataRefetch}
                message="Error in fetching data"
                heading="My Compaints"
                headingClassName="text-xl md:text-2xl font-medium mx-8 my-4 mb-4"
              />
            </Paper>
          }
        >
          <Paper className="w-full pb-8 h-[50vh] md:w-[48%] lg:w-1/2 px-0  md:h-full flex-grow overflow-hidden rounded-xl">
            <Typography className="mx-8 my-4 mb-2.5 text-xl md:text-2xl font-medium ">
              My Complaints
            </Typography>
            <TableComponent
              columns={complaintTablecolumns}
              isLoading={isLoadingComplaintData}
              rows={complaintData || []}
              tableClassName="px-4 pb-4 h-full border-0"
              onRowClick={() => {
                navigate("complaints");
              }}
            />
          </Paper>
        </ErrorBoundary>
      </Box>
      {isOpen && (
        <AlertComponent
          message={message}
          severity={severity}
          handleClose={() => handleAlert(false)}
        />
      )}
    </Box>
  );
};

export default StudentHome;
