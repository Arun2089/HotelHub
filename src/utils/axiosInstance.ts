import axios, { AxiosInstance } from "axios";
import { BASE_URL } from "@constant/index";
import { deleteLocalStorage, getLocalStorage } from "@utils/index";

const axiosInstance = (): AxiosInstance => {
  const instance = axios.create({
    baseURL: BASE_URL,
  });
  instance.interceptors.request.use(
    (config) => {
      const token = getLocalStorage("jwtToken");
      if (token) config.headers.Authorization = `Bearer ${token}`;
      else window.location.href = "/login";
      return config;
    },
    (error) => Promise.reject(error)
  );

  instance.interceptors.response.use(
    (response) => response,
    (error) => {
      if (error.response.status === 401) {
        deleteLocalStorage("jwtToken");
        window.location.href = "/login";
      }
      return Promise.reject(error);
    }
  );

  return instance;
};

export default axiosInstance;
