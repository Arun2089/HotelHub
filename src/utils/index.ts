import { StudentInfoType } from "@ts/types";
import { STUDENT_INFO_URL } from "@constant/index";

export const fetchData = async (url: string) => {
  const response = await fetch(url);
  if (response.ok) {
    const data = await response.json();
    return data;
  } else {
    throw new Error(`Error in retrieving data from ${url}`);
  }
};
export const sendData = async (
  url: string,
  method: "POST" | "PUT",
  data: any //eslint-disable-line
): Promise<boolean> => {
  const response = await fetch(url, {
    method: method,
    headers: {
      "Content-Type": "application/json",
    },
    body: JSON.stringify(data),
  });
  if (response.ok) {
    return true;
  } else {
    throw new Error(`Error in sending data`);
  }
};

export const deleteData = async (url: string): Promise<boolean> => {
  const response = await fetch(url, {
    method: "DELETE",
  });
  if (response.ok) {
    return true;
  } else {
    throw new Error(`Error in deleting data from ${url}`);
  }
};

export const checkEmailExists = async (
  email: string,
  id?: number
): Promise<boolean> => {
  try {
    const data = await fetchData(STUDENT_INFO_URL);
    if (id) {
      const emailExists = data.some(
        (student: StudentInfoType) =>
          student.email === email && student.id !== id
      );
      return emailExists;
    }
    const emailExists = data.some(
      (student: StudentInfoType) => student.email === email
    );
    return emailExists;
  } catch (error) {
    throw error as string;
  }
};

export const catchErrorMessage = (error: unknown) => {
  return error instanceof Error ? error.message : "An error occurred";
};
