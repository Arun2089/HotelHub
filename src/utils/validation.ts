function validateEmail(email: string) {
  const emailRegex = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;
  if (email.trim() === "" || !emailRegex.test(email)) {
    return false;
  }
  return true;
}

function validatePhone(phone: string) {
  const phoneRegex = /^[6-9]\d{9}$/;
  if (phone.trim() === "" || !phoneRegex.test(phone)) {
    return false;
  }
  return true;
}

function validatePassword(password: string) {
  if (password.length < 8 || password.trim() === "") {
    return false;
  }
  return true;
}

function validateText(text: string) {
  const nameRegex = /^[A-Za-z\s]+$/;
  if (text.trim() === "" || !nameRegex.test(text)) {
    return false;
  }
  return true;
}

const validateError = (value: string, type: string) => {
  if (value === "") return false;
  switch (type) {
    case "text":
      return !validateText(value);
    case "email":
      return !validateEmail(value);
    case "tel":
      return !validatePhone(value);
    case "password":
    case "confirmPassword":
      return !validatePassword(value);
    default:
      return false;
  }
};
const getHelperText = (
  value: string,
  type: string,
  confirmPassword?: string
): string => {
  if (validateError(value, type)) {
    switch (type) {
      case "text":
        return "Name cannot have special characters";
      case "email":
        return "Enter a valid email";
      case "tel":
        return "Enter a valid phone number";
      case "password":
      case "confirmPassword": {
        return "Password must contain atleast 8 characters";
      }
      default:
        return "Invalid input";
    }
  } else if (type === "confirmPassword" && value && value !== confirmPassword) {
    return "Password does not match";
  } else return "";
};

export {
  validateEmail,
  validatePhone,
  validatePassword,
  validateText,
  validateError,
  getHelperText,
};
