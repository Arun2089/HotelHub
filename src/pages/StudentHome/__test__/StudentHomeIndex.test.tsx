import { render, screen, waitFor, fireEvent } from "@testing-library/react";
import "@testing-library/jest-dom";
import StudentHome from "..";
import { fetchData } from "@src/utils";

jest.mock("@src/utils/index.ts", () => {
  const BASE_URL = "http://localhost:1337";
  const NOTICES_URL: string = BASE_URL + "/api/notices";
  const COMPLAINTS_URL = BASE_URL + "/api/complaints";
  const TODAY_MENU_URL = BASE_URL + "/api/todaymenu";
  const ROOM_INFO_URL = BASE_URL + "/api/room-info";

  return {
    ...jest.requireActual("@src/utils/index.ts"),
    fetchData: jest.fn().mockImplementation((url) => {
      switch (url) {
        case NOTICES_URL:
          return Promise.resolve({
            data: [
              {
                id: 68,
                attributes: {
                  title: "holiday",
                  date: "2023-12-22",
                  content: "ezxfyguhijk",
                  createdAt: "2023-12-22T13:34:14.434Z",
                  updatedAt: "2023-12-22T13:34:14.434Z",
                  publishedAt: "2023-12-22T13:34:14.433Z",
                },
              },
              {
                id: 69,
                attributes: {
                  title: "jnfwoehfouwehf",
                  date: "2023-12-22",
                  content: "wfpijwerifjwirojfwrf",
                  createdAt: "2023-12-22T13:34:20.929Z",
                  updatedAt: "2023-12-22T13:34:20.929Z",
                  publishedAt: "2023-12-22T13:34:20.928Z",
                },
              },
            ],
          });
        case ROOM_INFO_URL:
          return Promise.resolve({
            data: {
              id: 1,
              attributes: {
                data: {
                  Wing: "A",
                  Floor: 1,
                  Room: 101,
                },
              },
            },
          });
        case COMPLAINTS_URL:
          return Promise.resolve({
            data: [
              {
                id: 1,
                attributes: {
                  type: "Complaint 1",
                  date: "2022-01-01",
                  description: "This is complaint 1",
                  status: "pending",
                  studentName: "Test Name 1",
                },
              },
              {
                id: 2,
                attributes: {
                  type: "Complaint 2",
                  date: "2022-01-02",
                  description: "This is complaint 2",
                  status: "resolved",
                  studentName: "Test Name 2",
                },
              },
              {
                id: 3,
                attributes: {
                  type: "Complaint 3",
                  date: "2022-01-03",
                  description: "This is complaint 3",
                  status: "invalid",
                  studentName: "Test Name 3",
                },
              },
            ],
          });
        case TODAY_MENU_URL:
          return Promise.resolve({
            data: {
              id: 1,
              attributes: {
                data: [
                  {
                    name: "Breakfast",
                    value: "Idli, Vada, Sambar, Chutney",
                  },
                  {
                    name: "Lunch",
                    value: "Rice, Sambar, Rasam, Poriyal, Appalam",
                  },
                  {
                    name: "Dinner",
                    value: "Chapathi, Dal, Poriyal, Curd, Pickle",
                  },
                ],
              },
            },
          });

        default:
          return Promise.reject(new Error("not found"));
      }
    }),
  };
});

describe("StudentHome", () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  it("snapshot test", async () => {
    const { asFragment } = render(<StudentHome />);
    await waitFor(() => expect(asFragment()).toMatchSnapshot());
  });

  it("renders  Notices", async () => {
    render(<StudentHome />);
    await waitFor(() => {
      expect(screen.getByTestId("notice-component")).toBeInTheDocument();
    });
  });

  it("renders  Complaints", async () => {
    render(<StudentHome />);
    await waitFor(() => {
      expect(screen.getByText("Complaint 1")).toBeInTheDocument();
      expect(screen.getByText("Complaint 2")).toBeInTheDocument();
      expect(screen.getByText("Complaint 3")).toBeInTheDocument();
    });
  });

  it("renders  Today's Menu", async () => {
    render(<StudentHome />);
    await waitFor(() => {
      expect(screen.getByText("Breakfast")).toBeInTheDocument();
      expect(screen.getByText("Lunch")).toBeInTheDocument();
      expect(screen.getByText("Dinner")).toBeInTheDocument();
      expect(screen.getByText("Idli")).toBeInTheDocument();
      expect(screen.getByText("Rice")).toBeInTheDocument();
      expect(screen.getByText("Chapathi")).toBeInTheDocument();
    });
  });

  it("renders  Room Info", async () => {
    render(<StudentHome />);
    await waitFor(() => {
      expect(screen.getByText("A")).toBeInTheDocument();
      expect(screen.getByText("1")).toBeInTheDocument();
      expect(screen.getByText("101")).toBeInTheDocument();
    });
  });

  it("renders with error", async () => {
    (fetchData as jest.Mock).mockRejectedValue(
      new Error("something went wrong")
    );
    render(<StudentHome />);
    await waitFor(() => {
      expect(screen.getAllByText("Error in fetching data")).toHaveLength(4);
      expect(screen.getAllByRole("button", { name: "Reload" })).toHaveLength(4);
    });
  });

  it("onclick on reload button calls fetchData", async () => {
    (fetchData as jest.Mock).mockRejectedValue(
      new Error("something went wrong")
    );
    render(<StudentHome />);
    await waitFor(() => {
      expect(screen.getAllByText("Error in fetching data")).toHaveLength(4);
      expect(screen.getAllByRole("button", { name: "Reload" })).toHaveLength(4);
    });
    const reloadButton = screen.getAllByRole("button", { name: "Reload" })[0];
    fireEvent.click(reloadButton);
    expect(fetchData).toHaveBeenCalled();
  });
});
