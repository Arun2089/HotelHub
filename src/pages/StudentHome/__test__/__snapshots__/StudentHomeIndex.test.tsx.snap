// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StudentHome snapshot test 1`] = `
<DocumentFragment>
  <div
    class="p-4 w-full xl:p-8 flex flex-col min-h-20 flex-grow-0 md:flex-grow md:min-h-1/2 overflow-scroll MuiBox-root css-0"
  >
    <div
      class="flex-col gap-y-4 flex md:flex-row justify-around gap-x-6 md:gap-y-0 h-1/2 pb-6 MuiBox-root css-0"
    >
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 w-full pb-8 md:w-[48%] lg:w-1/2 px-0 h-1/2 md:h-full flex-grow overflow-hidden rounded-xl css-1ps6pg7-MuiPaper-root"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 mx-8 my-4 mb-2.5 text-xl md:text-2xl font-medium  css-ahj2mt-MuiTypography-root"
        >
          Room Information
        </p>
        <div
          class="flex justify-evenly px-4 gap-2 md:gap-4 h-32 md:h-40 mt-12 MuiBox-root css-0"
        >
          <span
            class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-153n0d3-MuiSkeleton-root"
            style="width: 100%; height: 100%;"
          />
        </div>
      </div>
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 w-full pb-8 md:w-[48%] lg:w-1/2 px-0 h-[50vh] md:h-full flex-grow overflow-hidden rounded-xl css-1ps6pg7-MuiPaper-root"
        data-testid="notice-component"
      >
        <div
          class="mx-8 my-4 mb-2.5 flex justify-between MuiBox-root css-0"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 text-xl md:text-2xl font-medium css-ahj2mt-MuiTypography-root"
          >
            Notice
          </p>
        </div>
        <div
          class="h-[90%] flex flex-col gap-y-4 justify-center items-center MuiBox-root css-0"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-colorPrimary MuiSvgIcon-fontSizeMedium w-20 h-20 css-ci5apu-MuiSvgIcon-root"
            data-testid="SmsFailedOutlinedIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M20 2H4c-1.1 0-2 .9-2 2v18l4-4h14c1.1 0 2-.9 2-2V4c0-1.1-.9-2-2-2zm0 14H5.17L4 17.17V4h16v12zm-9-4h2v2h-2zm0-6h2v4h-2z"
            />
          </svg>
          <p>
            No notices
          </p>
        </div>
      </div>
    </div>
    <div
      class="flex-col gap-y-4 flex md:flex-row justify-around gap-x-6 md:gap-y-0 h-1/2 pb-3 MuiBox-root css-0"
    >
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 w-full md:w-1/2 h-full  rounded-xl css-1ps6pg7-MuiPaper-root"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 mx-8 my-4 mb-2.5 text-xl md:text-2xl font-medium  css-ahj2mt-MuiTypography-root"
        >
          Today's Menu
        </p>
        <div
          class="flex justify-evenly mx-4 py-4 gap-4 max-h-fit mt-8 mb-6 border rounded-xl MuiBox-root css-0"
        >
          <span
            class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-153n0d3-MuiSkeleton-root"
            style="width: 100%; height: 100%;"
          />
        </div>
      </div>
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 w-full pb-8 h-[50vh] md:w-[48%] lg:w-1/2 px-0  md:h-full flex-grow overflow-hidden rounded-xl css-1ps6pg7-MuiPaper-root"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 mx-8 my-4 mb-2.5 text-xl md:text-2xl font-medium  css-ahj2mt-MuiTypography-root"
        >
          My Complaints
        </p>
        <div
          aria-colcount="4"
          aria-multiselectable="false"
          aria-rowcount="101"
          class="px-4 pb-4 h-full border-0 MuiDataGrid-root MuiDataGrid-root--densityStandard MuiDataGrid-withBorderColor css-13dniha-MuiDataGrid-root"
          role="grid"
        >
          <div
            class="MuiDataGrid-main css-204u17-MuiDataGrid-main"
          >
            <div
              class="MuiDataGrid-columnHeaders MuiDataGrid-withBorderColor css-1iyq7zh-MuiDataGrid-columnHeaders"
              role="presentation"
              style="min-height: 56px; max-height: 56px; line-height: 56px;"
            >
              <div
                class="MuiDataGrid-columnHeadersInner MuiDataGrid-columnHeadersInner--scrollable css-gl260s-MuiDataGrid-columnHeadersInner"
                role="rowgroup"
                style="transform: translate3d(0px, 0px, 0px);"
              >
                <div
                  aria-rowindex="1"
                  class="css-yrdy0g-MuiDataGrid-columnHeaderRow"
                  role="row"
                >
                  <div
                    aria-colindex="1"
                    aria-sort="none"
                    class="MuiDataGrid-columnHeader MuiDataGrid-columnHeader--alignCenter MuiDataGrid-withBorderColor pl-8"
                    data-field="id"
                    role="columnheader"
                    style="height: 56px; width: 100px; min-width: 100px; max-width: 100px;"
                    tabindex="0"
                  >
                    <div
                      class="MuiDataGrid-columnHeaderDraggableContainer"
                      draggable="false"
                      role="presentation"
                    >
                      <div
                        class="MuiDataGrid-columnHeaderTitleContainer"
                        role="presentation"
                      >
                        <div
                          class="MuiDataGrid-columnHeaderTitleContainerContent"
                        >
                          <p
                            class="MuiTypography-root MuiTypography-body1 font-semibold css-ahj2mt-MuiTypography-root"
                          >
                            Id
                          </p>
                        </div>
                      </div>
                    </div>
                    <div
                      class="MuiDataGrid-columnSeparator MuiDataGrid-columnSeparator--sideRight"
                      style="min-height: 56px; opacity: 1;"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiDataGrid-iconSeparator css-i4bv87-MuiSvgIcon-root"
                        data-testid="SeparatorIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M11 19V5h2v14z"
                        />
                      </svg>
                    </div>
                  </div>
                  <div
                    aria-colindex="2"
                    aria-sort="none"
                    class="MuiDataGrid-columnHeader MuiDataGrid-columnHeader--alignCenter MuiDataGrid-withBorderColor pl-8"
                    data-field="date"
                    role="columnheader"
                    style="height: 56px; width: 150px; min-width: 150px; max-width: 150px;"
                    tabindex="-1"
                  >
                    <div
                      class="MuiDataGrid-columnHeaderDraggableContainer"
                      draggable="false"
                      role="presentation"
                    >
                      <div
                        class="MuiDataGrid-columnHeaderTitleContainer"
                        role="presentation"
                      >
                        <div
                          class="MuiDataGrid-columnHeaderTitleContainerContent"
                        >
                          <p
                            class="MuiTypography-root MuiTypography-body1 font-semibold css-ahj2mt-MuiTypography-root"
                          >
                            Date
                          </p>
                        </div>
                      </div>
                    </div>
                    <div
                      class="MuiDataGrid-columnSeparator MuiDataGrid-columnSeparator--sideRight"
                      style="min-height: 56px; opacity: 1;"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiDataGrid-iconSeparator css-i4bv87-MuiSvgIcon-root"
                        data-testid="SeparatorIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M11 19V5h2v14z"
                        />
                      </svg>
                    </div>
                  </div>
                  <div
                    aria-colindex="3"
                    aria-sort="none"
                    class="MuiDataGrid-columnHeader MuiDataGrid-columnHeader--alignCenter MuiDataGrid-withBorderColor"
                    data-field="type"
                    role="columnheader"
                    style="height: 56px; width: 0px; min-width: 0; max-width: 0;"
                    tabindex="-1"
                  >
                    <div
                      class="MuiDataGrid-columnHeaderDraggableContainer"
                      draggable="false"
                      role="presentation"
                    >
                      <div
                        class="MuiDataGrid-columnHeaderTitleContainer"
                        role="presentation"
                      >
                        <div
                          class="MuiDataGrid-columnHeaderTitleContainerContent"
                        >
                          <p
                            class="MuiTypography-root MuiTypography-body1 font-semibold css-ahj2mt-MuiTypography-root"
                          >
                            Type
                          </p>
                        </div>
                      </div>
                    </div>
                    <div
                      class="MuiDataGrid-columnSeparator MuiDataGrid-columnSeparator--sideRight"
                      style="min-height: 56px; opacity: 1;"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiDataGrid-iconSeparator css-i4bv87-MuiSvgIcon-root"
                        data-testid="SeparatorIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M11 19V5h2v14z"
                        />
                      </svg>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="MuiDataGrid-virtualScroller css-qvtrhg-MuiDataGrid-virtualScroller"
              role="presentation"
              style="overflow-x: hidden;"
            >
              <div
                class="MuiDataGrid-overlayWrapper css-19qgr0z-MuiDataGrid-overlayWrapper"
              >
                <div
                  class="MuiDataGrid-overlayWrapperInner css-1akuw9y-MuiDataGrid-overlayWrapperInner"
                  style="height: 0px; width: 0px;"
                >
                  <div
                    class="MuiDataGrid-overlay css-1w53k9d-MuiDataGrid-overlay"
                  >
                    <span
                      class="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary css-18lrjg1-MuiCircularProgress-root"
                      role="progressbar"
                      style="width: 40px; height: 40px;"
                    >
                      <svg
                        class="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
                        viewBox="22 22 44 44"
                      >
                        <circle
                          class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-176wh8e-MuiCircularProgress-circle"
                          cx="44"
                          cy="44"
                          fill="none"
                          r="20.2"
                          stroke-width="3.6"
                        />
                      </svg>
                    </span>
                  </div>
                </div>
              </div>
              <div
                class="MuiDataGrid-virtualScrollerContent MuiDataGrid-virtualScrollerContent--overflowed css-1kwdphh-MuiDataGrid-virtualScrollerContent"
                role="presentation"
                style="width: auto; height: 1px; min-height: auto;"
              >
                <div
                  class="MuiDataGrid-virtualScrollerRenderZone css-s1v7zr-MuiDataGrid-virtualScrollerRenderZone"
                  role="rowgroup"
                  style="transform: translate3d(0px, undefinedpx, 0px);"
                />
              </div>
            </div>
          </div>
          <div
            class="MuiDataGrid-footerContainer MuiDataGrid-withBorderColor css-wop1k0-MuiDataGrid-footerContainer"
          >
            <div />
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
